### PR TITLE
Update privatetunnel from 2.8 to 3.0.1.1643

### PR DESCRIPTION
**Pull request by suschizu**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

After making all changes to a cask, verify:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` left no offenses.
- [ ] There are no [open pull requests](https://github.com/Homebrew/homebrew-cask/pulls) for the same update.
- [ ] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#stable-versions) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

### CODE REVIEW

File: Casks/privatetunnel.rb [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
+  # swupdate.openvpn.net/downloads/privatetunnel/ was verified as official when first introduced to the cask
+  url "https://swupdate.openvpn.net/downloads/privatetunnel/private-tunnel-#{version.before_comma}.#{version.after_comma}_signed.dmg"
   appcast "https://www.privatetunnel.com/apps/"
   name "Private Tunnel"
   homepage "https://www.privatetunnel.com/home/"
 
-  pkg "PrivateTunnel.pkg"
+  pkg "Private_Tunnel_#{version.before_comma.dots_to_underscores}(#{version.after_comma})_Installer_signed.pkg"
 
   uninstall script: "Uninstaller.command"
```
Comment (vitorgalvao) [inline Casks/privatetunnel.rb:13]: Looks like this no longer exists.

Comment (suschizu) [reply Casks/privatetunnel.rb:13]: hm strange - I've downloaded this version yesterday but it#s gone today

Comment (core-code) [reply Casks/privatetunnel.rb:13]: are you sure? i can confirm this download location from the website. all the travis failures are about uninstall

https://swupdate.openvpn.net/downloads/privatetunnel/private-tunnel-3.0.1.1643_signed.dmg

Comment (miccal) [inline Casks/privatetunnel.rb:13]: Try this:
```
  uninstall script: {
    executable: "/Applications/Private Tunnel/Uninstall Private Tunnel.app/Contents/Resources/remove.sh",
    sudo:       true,
  }
```

**Comment by core-code** [comment]
```
==> Purging files for version 3.0.1,1643 of Cask privatetunnel
Error: Some packages are still installed, add them to uninstall pkgutil:
org.openvpn.privatetunnel.pkg
org.openvpn.privatetunnel_framework.pkg
org.openvpn.privatetunnel_launch.pkg
org.openvpn.privatetunnel_uninstall.pkg
Error: Some launch jobs are still installed, add them to uninstall launchctl:
org.openvpn.privatetunnel
##[error]Error: Some packages are still installed, add them to uninstall pkgutil:
org.openvpn.privatetunnel.pkg
org.openvpn.privatetunnel_framework.pkg
org.openvpn.privatetunnel_launch.pkg
org.openvpn.privatetunnel_uninstall.pkg

Error: Some launch jobs are still installed, add them to uninstall launchctl:
org.openvpn.privatetunnel
```

**Comment by miccal** [comment]
Ok, try this one instead:
```
  uninstall script: {
    executable: "/Applications/Private Tunnel/Uninstall Private Tunnel.app/Contents/MacOS/uninstaller",
    sudo:       true,
  }
```

File: Casks/privatetunnel.rb [hunk 2]
```diff
@@ -1,14 +1,17 @@
...
   appcast "https://www.privatetunnel.com/apps/"
   name "Private Tunnel"
   homepage "https://www.privatetunnel.com/home/"
 
-  pkg "PrivateTunnel.pkg"
+  pkg "Private_Tunnel_#{version.before_comma.dots_to_underscores}(#{version.after_comma})_Installer_signed.pkg"
 
-  uninstall script: "Uninstaller.command"
+  uninstall script: {
+    executable: "/Applications/Private Tunnel/Uninstall Private Tunnel.app/Contents/MacOS/uninstaller",
```
Comment (vitorgalvao) [inline Casks/privatetunnel.rb:14]: Doesn’t this invoke a GUI?

Comment (miccal) [reply Casks/privatetunnel.rb:14]: Yes it does.

**Comment by miccal** [comment]
Contents of `remove.sh`, which should help in creating a working `uninstall` and `zap`:
<details>
<summary>remove.sh</summary>

```
#!/usr/bin/env bash

LOG="/tmp/privatetunnel-uninstall.log"
exec > $LOG 2>&1

FRAMEWORK="/Library/Frameworks/PrivateTunnelV3.framework"
LAUNCH_DAEMON="/Library/LaunchDaemons/openvpn.net.privatetunnel.plist"
LAUNCH_DAEMON2="/Library/LaunchDaemons/org.openvpn.privatetunnel.plist"

APP="/Applications/Private Tunnel"

APP_SUPP=~/Library/"Application Support"/"Private Tunnel"
LOG="/tmp/openvpn-uninstall.log"

CACHES=~/Library/"Saved Application State"/"openvpn.net.privatetunnel.app.savedState"
PREFS1=~/Library/"Preferences"/"openvpn.net.privatetunnel.app.helper.plist"
PREFS2=~/Library/"Preferences"/"openvpn.net.privatetunnel.app.plist"
LOGS=~/Library/"Logs"/"Private Tunnel"

APP_LINK="/Applications/Private Tunnel.app"


# close app
echo "Closing app..."
osascript -e 'quit app "Private Tunnel"'
echo ""


echo "Services:"
launchctl list | grep openvpn
echo ""
echo "Daemon property lists:"
ls /Library/LaunchDaemons | grep openvpn
echo ""


echo "Removing openvpn.net.privatetunnel..."
launchctl stop openvpn.net.privatetunnel
launchctl unload /Library/LaunchDaemons/openvpn.net.privatetunnel.plist
echo ""
echo "Removing org.openvpn.privatetunnel..."
launchctl stop org.openvpn.privatetunnel
launchctl unload /Library/LaunchDaemons/org.openvpn.privatetunnel.plist
echo ""


echo "Services:"
launchctl list | grep openvpn
echo ""
echo "Daemon property lists:"
ls /Library/LaunchDaemons | grep openvpn
echo ""


# remove everything, but preserve daemon plist to unload it on next install
echo "Cleaning up..."
rm -rf "$APP" "$APP_SUPP" "$FRAMEWORK" "$CACHES" "$PREFS1" "$PREFS2" "$LOGS" "$APP_LINK"
echo ""

echo "uninstall finished"
```
</details>

**Pull request merged**